### PR TITLE
GLOCKFILE: update deps

### PR DESCRIPTION
**Pull request by tamird**
Fixes a bunch of issues found by ineffassign.
- [x] @tschottdorf: PTAL server/admin.go
- [x] @paperstreet: PTAL sql/backfill.go
- [x] @dt: PTAL sql/check.go
- [x] @mberhault: PTAL sql/config_test.go
- [x] @raduberinde: PTAL sql/index_selection.go
- [x] @dt: PTAL sql/pgwire/types.go
- [x] @andreimatei: PTAL sql/rename_test.go
- [x] @paperstreet: PTAL sql/sqlbase/table.go
- [x] @jordanlewis: PTAL storage/store.go

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9695)

### CODE REVIEW

**Comment by tschottdorf** [comment]
:lgtm: on server/admin.go

---

Reviewed 16 of 16 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9695#-:-KTBuS08WNyQvxuxbITp:bnj63kx)*

File: sql/index_selection.go [hunk 1]
```diff
@@ -380,7 +380,7 @@ func (v *indexInfo) analyzeOrdering(
 	// Analyze the ordering provided by the index (either forward or reverse).
 	fwdIndexOrdering := scan.computeOrdering(v.index, v.exactPrefix, false)
 	revIndexOrdering := scan.computeOrdering(v.index, v.exactPrefix, true)
-	fwdMatch, orderCols := analyzeOrdering(fwdIndexOrdering)
+	fwdMatch, _ := analyzeOrdering(fwdIndexOrdering) // TODO(radu): why is the second return being ignored?
```
Comment (RaduBerinde) [inline sql/index_selection.go]: the `orderCols` return indicates how many total columns a desired ordering has so it should be independent of the argument

**Comment by RaduBerinde** [comment]
LGTM

File: sql/index_selection.go (hunk 1 above)
Comment (tamird) [inline sql/index_selection.go]: Updated.

File: sql/config_test.go [hunk 2]
```diff
@@ -92,7 +92,7 @@ func TestGetZoneConfig(t *testing.T) {
 	defer srv.Stopper().Stop()
 	s := srv.(*server.TestServer)
 
-	expectedCounter := uint32(keys.MaxReservedDescID + 1)
+	expectedCounter := uint32(keys.MaxReservedDescID)
```
Comment (mberhault) [inline sql/config_test.go]: out of curiosity, why does this fail?

**Review by mberhault — APPROVED** [review]

**Review by dt — APPROVED** [review]
fix in `check.go` lgtm

File: sql/config_test.go (hunk 2 above)
Comment (tamird) [inline sql/config_test.go]: It doesn't. What fails is the pattern of post-incrementing, which results in the final increment being ineffectual.

**Comment by andreimatei** [comment]
:lgtm: on `rename_test.go`

---

Review status: 14 of 16 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9695#-:-KTF5kc3Ua-RYdkiTxhr:btq5ypb)*

**Comment by paperstreet** [comment]
:lgtm: for backfill.go and table.go

---

Review status: 14 of 16 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9695#-:-KTF6FhgmH-YN0dav-nI:b-dpi8mt)*

**Pull request merged**